src/addressBook/index.js

```diff
@@ -0,0 +1,339 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable camelcase */
+
+import {
+  addStringUnlocalized,
+  addUrl,
+  getStringNoLocale,
+  getStringNoLocaleAll,
+  getUrl,
+  getUrlAll,
+  setThing,
+} from "@inrupt/solid-client";
+import { v4 as uuid } from "uuid";
+import { rdf, dc, acl, vcard, ldp, foaf } from "rdf-namespaces";
+import {
+  createResponder,
+  defineDataset,
+  defineThing,
+  isHTTPError,
+  ERROR_CODES,
+} from "../solidClientHelpers/utils";
+import { getResource, saveResource } from "../solidClientHelpers/resource";
+import { joinPath } from "../stringHelpers";
+
+const CONTACTS_CONTAINER = "contacts/";
+
+export function vcardExtras(property) {
+  return `http://www.w3.org/2006/vcard/ns#${property}`;
+}
+
+export function contactsContainerIri(iri) {
+  return joinPath(iri, CONTACTS_CONTAINER);
+}
+
+export function createAddressBook({ iri, owner, title = "Contacts" }) {
+  const indexIri = joinPath(iri, "index.ttl");
+  const peopleIri = joinPath(iri, "people.ttl");
+  const groupsIri = joinPath(iri, "groups.ttl");
+
+  const index = defineDataset(
+    { name: "this" },
```

Comment:
@megoth This line here should be naming the index `this`. I'm not sure why it would not honor this setting.

Reply:
Yeah, it's a weird bug.

I just tested it again, and the payload sent to create is:

```
<#16000959209144584020461100542> a <http://www.w3.org/2006/vcard/ns#AddressBook>;
    <http://www.w3.org/ns/auth/acl#owner> <https://megoth-demo1.inrupt.net/profile/card#me>;
    <http://purl.org/dc/elements/1.1/title> "Contacts";
    <http://www.w3.org/2006/vcard/ns#nameEmailIndex> <https://megoth-demo1.inrupt.net/contacts/people.ttl>;
    <http://www.w3.org/2006/vcard/ns#groupIndex> <https://megoth-demo1.inrupt.net/contacts/groups.ttl>.
```

Could it be a bug in the Solid client?